jax/lax/lax.py

```diff
@@ -3599,9 +3599,9 @@ def _slice_shape_rule(operand, *, start_indices, limit_indices, strides):
       msg = "slice strides must be positive, got {}"
       raise TypeError(msg.format(strides))
 
-  result_shape = np.floor_divide(
-      np.add(np.subtract(limit_indices, start_indices), strides) - 1, strides)
-  return tuple(result_shape)
+  diff = np.subtract(limit_indices, start_indices)
+  # Not np.divmod since Poly.__rdivmod__ is ignored by NumPy, breaks poly stride
+  return tuple(q + np.where(r, 1, 0) for q, r in map(divmod, diff, strides))
```

Comment:
I was initially confused by the use of `np.where`, thinking that it is going to break in presence of polynomials. But `r` is always a scalar, right? Then perhaps `(r > 0)` is cleaner than `np.where(r, 1, 0)`?

Reply:
Good point, updated to `r > 0`!